bookkeeper-server/src/main/java/org/apache/bookkeeper/net/NetUtils.java

```diff
@@ -85,4 +92,46 @@ public static String resolveNetworkLocation(DNSToSwitchMapping dnsResolver, Inet
         return rNames.get(0);
     }
 
+    public static String getLocalHost() throws UnknownHostException {
+        String ip = InetAddress.getLocalHost().getHostAddress();
+        if (InetAddress.getLocalHost().isLoopbackAddress()) {
+            return getAddress();
+        } else {
+            return ip;
+        }
+
+    }
+
+    private static boolean isboolIp(String ipAddress) {
+        Matcher matcher = pattern.matcher(ipAddress);
+        return matcher.matches();
+    }
+
+    private static String getAddress() throws UnknownHostException {
```

Comment:
please call this method **getValidAddress** or something like that

Reply:
Already dealt with at  910f6c7